plutus-use-cases/src/Language/PlutusTx/Coordination/Contracts/PubKey.hs

```diff
@@ -52,16 +51,19 @@ pubKeyContract pk vl = do
     tid <- writeTxSuccess tx
 
     ledgerTx <- awaitTransactionConfirmed address tid
-    let output = listToMaybe
-                $ fmap fst
-                $ filter ((==) address . txOutAddress . snd)
-                $ Map.toList
+    let output = Map.keys
+                $ Map.filter ((==) address . txOutAddress)
                 $ unspentOutputsTx ledgerTx
     ref <- case output of
-        Nothing -> throwing _OtherError $
+        [] -> throwing _OtherError $
             "Transaction did not contain script output"
             <> "for public key '"
             <> T.pack (show pk)
             <> "'"
-        Just o -> pure $ scriptTxIn o (pkValidator pk) (RedeemerScript $ PlutusTx.toData ())
+        [o] -> pure $ scriptTxIn o (pkValidator pk) (RedeemerScript $ PlutusTx.toData ()) (DataScript $ PlutusTx.toData ())
+        _ -> throwing _OtherError $
+            "Transaction did contained multiple script outputs"
```

Comment:
Without the `did`? Maybe print those outputs to the user?

Reply:
Me being lazy.